### ISSUE TITLE
dead code in ck_sort

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11659]
Migrated from [rt.perl.org#99426](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=99426) (status was 'resolved')

Searchable as RT99426$

**Comment by p5pRT** [comment]
### From @cpansprout

This commit added support for sort $coderef @&#8203;foo&#8203;:

commit 7a4920e67d1e2d67a4397a908141c6608866ebb0
Author&#8203;: Graham Barr \<gbarr@&#8203;pobox\.com>
Date&#8203;:   Fri Nov 27 05&#8203;:16&#8203;:50 1998 \+0000

&nbsp;    integrate change\#2246 from mainline\, while still allowing
&nbsp;    C\<sort $globref @&#8203;foo>
&nbsp;    
&nbsp;    allow C\<sort $coderef @&#8203;foo>
&nbsp;    
&nbsp;    p4raw\-link&#8203;: @&#8203;2246 on //depot/perl&#8203;: c6e96bcb406bc8b8d8610606459ff606ad6883aa
&nbsp;    
&nbsp;    p4raw\-id&#8203;: //depot/maint\-5\.005/perl@&#8203;2315
&nbsp;    p4raw\-integrated&#8203;: from //depot/perl@&#8203;2314 'merge in' t/op/sort\.t
&nbsp;        \(@&#8203;1760\.\.\)

But this part of it I don’t understand&#8203;:

<details><summary>Inline Patch</summary>

```diff
diff --git a/op.c b/op.c
index fcb5f0e..c991051 100644
--- a/op.c
+++ b/op.c
@@ -4768,7 +4768,9 @@ ck_sort(OP *o)
 		kid->op_next = k;
 	    o->op_flags |= OPf_SPECIAL;
 	}
-    }
+	else if (kid->op_type == OP_RV2SV || kid->op_type == OP_PADSV)
+	    null(cLISTOPo->op_first->op_sibling);
+     }
 
     return o;
 }


```
</details>
Nowadays, the op that it is trying to null is already a null, so presumably toke.c or perly.y has changed since then.  So I think this else block can be deleted.

The null I’m referring to is

$ perl \-MO=Concise \-e 'sort $fo @&#8203;fo'
8  \<@&#8203;> leave\[1 ref\] vKP/REFC \->\(end\)
1     \<0> enter \->2
2     \<;> nextstate\(main 1 \-e&#8203;:1\) v&#8203;:\{ \->3
7     \<@&#8203;> sort vKS \->8
3        \<0> pushmark s \->4
\-        \<1> null K/1 \->5                  \<\-\- over here\.
\-           \<1> ex\-rv2sv sK/1 \->\-
4              \<\#> gvsv\[\*fo\] s \->5
6        \<1> rv2av\[t3\] lK/1 \->7
5           \<\#> gv\[\*fo\] s \->6
\-e syntax OK



---
Flags&#8203;:
&nbsp;    category=core
&nbsp;    severity=low


---
Site configuration information for perl 5\.15\.2&#8203;:

Configured by sprout at Wed Sep 14 14&#8203;:21&#8203;:56 PDT 2011\.

Summary of my perl5 \(revision 5 version 15 subversion 2\) configuration&#8203;:
&nbsp;  Snapshot of&#8203;: 828d619543f6e6d3ccfdad45caa681f9105b651a
&nbsp;  Platform&#8203;:
&nbsp;    osname=darwin\, osvers=10\.5\.0\, archname=darwin\-thread\-multi\-ld\-2level
&nbsp;    uname='darwin pint\.local 10\.5\.0 darwin kernel version 10\.5\.0&#8203;: fri nov 5 23&#8203;:20&#8203;:39 pdt 2010; root&#8203;:xnu\-1504\.9\.17~1release\_i386 i386 '
&nbsp;    config\_args='\-de \-Dusedevel \-Ui\_db \-Duseithreads \-Duselongdouble \-DDEBUGGING=\-g'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=undef\, use64bitall=undef\, uselongdouble=define
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'\,
&nbsp;    optimize='\-O3 \-g'\,
&nbsp;    cppflags='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.2\.1 \(Apple Inc\. build 5664\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='long double'\, nvsize=16\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='env MACOSX\_DEPLOYMENT\_TARGET=10\.3 cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib
&nbsp;    libs=\-ldbm \-ldl \-lm \-lutil \-lc
&nbsp;    perllibs=\-ldl \-lm \-lutil \-lc
&nbsp;    libc=\, so=dylib\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=bundle\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags=' \-bundle \-undefined dynamic\_lookup \-L/usr/local/lib \-fstack\-protector'

Locally applied patches&#8203;:
&nbsp;    



---
@&#8203;INC for perl 5\.15\.2&#8203;:
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.15\.2/darwin\-thread\-multi\-ld\-2level
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.15\.2
&nbsp;    /usr/local/lib/perl5/5\.15\.2/darwin\-thread\-multi\-ld\-2level
&nbsp;    /usr/local/lib/perl5/5\.15\.2
&nbsp;    /usr/local/lib/perl5/site\_perl
&nbsp;    \.



---
Environment for perl 5\.15\.2&#8203;:
&nbsp;    DYLD\_LIBRARY\_PATH \(unset\)
&nbsp;    HOME=/Users/sprout
&nbsp;    LANG=en\_US\.UTF\-8
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/usr/bin&#8203;:/bin&#8203;:/usr/sbin&#8203;:/sbin&#8203;:/usr/local/bin&#8203;:/usr/X11/bin&#8203;:/usr/local/bin
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed by 3b71a32680\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'new' to 'resolved'
